core/tests/test_utils.py

```diff
@@ -809,9 +942,22 @@ def __get__(self, instance, owner):
         return self
 
     def pre_call_hook(self, args):
+        """Override this to do tasks that should be executed before the
+        actual function call.
+
+        Args:
+            args: list(str). Set of arguments that the function accepts.
```

Comment:
See my previous comment here. Can you explain why this is str and not *?

Reply:
This function takes a list of args of any type. Therefore, its type should be list(*) and not str.